File: backend/src/main/java/com/github/marceloleite2604/pitanga/model/event/checkroomexists/CheckRoomExists.java
package com.github.marceloleite2604.pitanga.model.event.checkroomexists;

import com.github.marceloleite2604.pitanga.model.event.Event;
import com.github.marceloleite2604.pitanga.model.event.EventType;
import lombok.Builder;

public class CheckRoomExists extends Event<CheckRoomExistsPayload> {

    public CheckRoomExists() {
        this(null);
    }

    @Builder
    public CheckRoomExists(CheckRoomExistsPayload payload) {
        super(EventType.CHECK_ROOM_EXISTS, payload);
    }
}

File: frontend/src/shared/model/events/CheckRoomExists.ts
import { RoomDto } from '..';
import { Event } from './Event';
import { EventType } from './EventType';

export interface CheckRoomExistsPayload {
  room: RoomDto,
  exists?: boolean
}

export interface CheckRoomExistsEvent extends Event<EventType.CheckRoomExists, CheckRoomExistsPayload> { }

export function buildCheckRoomExists(room: RoomDto) {
  return {
    type: EventType.CheckRoomExists,
    payload: {
      room
    }
  } as CheckRoomExistsEvent;
}
